training/training.module

```diff
@@ -442,26 +750,120 @@ function theme_test_form_theme_function($form) {
 }
 
 /**
+ * Creates array of users id from db.
+ *
+ * Select from table text_message users id who created text message.
+ *
+ * @param string $table
+ *   Table name.
+ * @param string $alias
+ *   Table alias.
+ *
+ * @return array $table
+ *   Returns array of users id.
+ */
+function _training_select_uid_from_db($table, $alias) {
+  $uid = array();
+  $query = db_select($table, $alias);
+  $query
+    ->fields($alias, array('uid'))
+    ->distinct();
+  $result = $query->execute()->fetchAll();
+  foreach ($result as $value) {
+    $uid[] = $value->uid;
+  }
+
+  return $uid;
+}
+
+/**
  * Sends a request to the rates service.
  *
  * Parses the xml data. Converts xml object properties to string,
  * writes them to arrays. The resulting arrays sets in the variables.
+ *
+ * @param array $context
  */
-function _training_request_rates_service() {
+function _training_request_rates_service(&$context) {
   $date = '';
   $array_data = array();
   $request = drupal_http_request(TRAINING_BLOCK_RATES_SERVICE);
-  $data = new SimpleXMLElement($request->data);
-  foreach ($data->Currency as $value) {
-    $array_data[] = array(
-      'CharCode' => (string) $value->CharCode,
-      'Name' => (string) $value->Name,
-      'Rate' => (string) $value->Rate,
-    );
+  if (!empty($request)) {
+    $data = new SimpleXMLElement($request->data);
+    foreach ($data->Currency as $value) {
+      $array_data[] = array(
+        'CharCode' => (string) $value->CharCode,
```

Comment:
CharCode, Name and Rate values are already strings, no needs to use type casting here